app/components/subject-picker/component.js

```diff
@@ -1,25 +1,27 @@
 import Component from '@ember/component';
-import EmberObject from '@ember/object';
-import { computed } from '@ember/object';
+import EmberObject, { computed } from '@ember/object';
 import { sort, notEmpty } from '@ember/object/computed';
 import { A } from '@ember/array';
 import { inject as service } from '@ember/service';
 import $ from 'jquery';
 import Analytics from 'ember-osf/mixins/analytics';
 
 function arrayEquals(arr1, arr2) {
-    return arr1.length === arr2.length && arr1.reduce((acc, val, i) => acc && val === arr2[i], true);
+    return arr1.length === arr2.length
+        && arr1.reduce((acc, val, i) => acc && val === arr2[i], true);
 }
 
 function arrayStartsWith(arr, prefix) {
-    return prefix.reduce((acc, val, i) => acc && val && arr[i] && val.id === arr[i].id, true);
+    return prefix.reduce((acc, val, i) => acc && val
+        && arr[i] && val.id === arr[i].id, true);
 }
 
 const Column = EmberObject.extend({
-    sortDefinition: ['text:asc'],
+    sortDefinition: ['text:asc'], // eslint-disable-line ember/avoid-leaking-state-in-components
```

Comment:
recollection that these were weird

Reply:
This can also be moved to `init()`.
[Rule](https://github.com/ember-cli/eslint-plugin-ember/blob/master/docs/rules/avoid-leaking-state-in-components.md): 
> Don't use arrays and objects as default properties.